runtime/src/qio/qio_popen.c

```diff
@@ -447,7 +454,14 @@ qioerr qio_proc_communicate(
   while( do_input || do_output || do_error ) {
     if( do_input ) {
       err = _qio_channel_flush_qio_unlocked(input);
-      if( !err ) do_input = false;
+      if( !err ) {
+        qioerr err2 = 0;
+        do_input = false;
+        // Close input channel.
+        err = qio_channel_close(false, input);
+        err2 = qio_file_close(input_file);
+        if( err2 && !err ) err = err2;
```

Comment:
I'm not so sure what to do with these errors, other that an error in one place but not both places needs to be returned. An error in qio_channel_close most probably means that there was an error actually writing the buffered data. That error can also occur in qio_file_close, depending on the OS's buffering strategy. But, in this context (where all the files are pipes) I don't know what would make such errors occur.